src/task_scl_structural_habitat.py

```diff
@@ -25,36 +25,344 @@ class str_hab(SCLTask):
         },
     }
 
-
-
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.land_cover, _ = self.get_most_recent_image(
             ee.ImageCollection(self.inputs["land_cover"]["ee_path"])
         )
         self.elevation = ee.Image(self.inputs["elevation"]["ee_path"])
         self.zones = ee.FeatureCollection(self.inputs["zones"]["ee_path"])
-        self.set_aoi_from_ee(self.inputs["probability"]["ee_path"])
-
+        # self.set_aoi_from_ee(self.inputs["probability"]["ee_path"])
 
     def calc(self):
 
-        print(land_cover)
+        zones_img = self.zones.reduceToImage(
+            properties=['Biome_zone'],
+            reducer=ee.Reducer.mode()
+        ).rename('Biome_zone')
+
+        def remap_elev(from_list, to_list, zone_number, z_img, elev_img, lc_img):
+            return (elev_img.lt(
+            lc_img.remap(from_list, to_list)
+                .updateMask(z_img.eq(zone_number)))
+                .multiply(100))
+
+
+        def remap_lc_by_zone(lc_list, z1_list, z2_list, z3_list, z_img, elev_img, lc_img):
```

Comment:
If it's not terribly hard, we should make this abstract, so that for a given species one could handle any number of zones. This assumes you know what's going to happen with the `reduceToImage` call above.